api/data.go

```diff
@@ -0,0 +1,67 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-legacy-cache-api/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+func (api *API) GetDataSets(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		log.Info(ctx, "Calling get datasets handler")
+
+		results, err := api.dataStore.GetDataSets(ctx)
+		if err != nil {
+			log.Error(ctx, "Error retrieving datasets from db:", err)
+			return
+		}
+
+		// Setting the header and encoding the results to JSON
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+		err = json.NewEncoder(w).Encode(results)
+		if err != nil {
+			log.Error(ctx, "Error encoding results to JSON: %v", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+func (api *API) AddDataSets(ctx context.Context) http.HandlerFunc {
+	log.Info(ctx, "api contains example endpoint, remove hello.go as soon as possible")
+
+	return func(w http.ResponseWriter, req *http.Request) {
+		// Deconstruct json into our models.DataMessage struct
+		var input models.DataMessage
+		err := json.NewDecoder(req.Body).Decode(&input)
+
+		if err != nil {
+			log.Error(ctx, "error decoding request body", err)
+			http.Error(w, "Bad Request", http.StatusBadRequest)
+			return
+		}
+
+		log.Info(ctx, "Received data - ", log.Data{"input:": input})
+
+		err = api.dataStore.AddDataSet(ctx, input)
+		if err != nil {
+			log.Error(ctx, "failed to insert document: %w", err)
```

Comment:
log events should never be dynamic, use a log.Data{} struct instead. In any case, this parameter is never populated as the event is not a formatted string
```suggestion
			log.Error(ctx, "failed to insert document", err)
```
This also applies to other log.Error lines in this service that contain a variable.